nixos/modules/services/network-filesystems/webdav-server-rs.nix

```diff
@@ -0,0 +1,109 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.webdav-server-rs;
+  format = pkgs.formats.toml { };
+  settings = recursiveUpdate
+    {
+      server.uid = config.users.users."${cfg.user}".uid;
+      server.gid = config.users.groups."${cfg.group}".gid;
```

Comment:
You might want to make this conditional on whether `cfg.user == "webdav"` and `cfg.group == "webdav"`. What if the user specified isn't declared in `configuration.nix`?

Reply:
This software requires uid and gid instead of user name and group name for default file permission.  That's a bad design.  However, making this optional will allow the daemon be root account and root group.  There's a big security risk.  I prefer to add assertions to ensure these attributes are defined.

Reply:
Sorry I sort of confused myself here thinking the user referenced was the user accessing `webdav`, not the user that `webdav` is being ran as. My apologies, you are definitely correct.

Reply:
I have added two assertions to ensure uid and gid are defined.